### PR TITLE
Upgrade to 'real' PF4 the react-consoles package

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

**Comment by KKoukiou** [comment]
OH of course, tests need some updates as well.

**Comment by martinpitt** [comment]
Lots of test failures still and conflicts, moving to draft.

File: package.json [hunk 1]
```diff
@@ -4,11 +4,11 @@
   "private": true,
   "dependencies": {
     "@patternfly/patternfly": "4.70.2",
-    "@patternfly/react-console": "2.0.16",
+    "@patternfly/react-console": "4.1.25",
     "@patternfly/react-core": "4.84.4",
     "@patternfly/react-table": "4.19.24",
     "@redhat/redhat-font": "git+https://github.com/RedHatOfficial/RedHatFont.git#2.2.0",
-    "@novnc/novnc": "1.1.0",
+    "@novnc/novnc": "1.2.0",
```
Comment (martinpitt) [inline package.json:11]: This should be dropped entirely again. Commit 76f48027 introduced it as a hack, but I think it's better to leave control of the novnc version to react-console.

**Review by martinpitt — APPROVED** [review]
Really nice cleanup, thank you! The unit-tests failure is because that branch wasn't rebased for some time, and is unrelated. I'm still re-triggering the integration tests as they look a bit sorry still.

I haven't tested this visually, but code-wise it looks great.

**Comment by martinpitt** [comment]
I visually tested this now in a fedora-33 VM, and the buttons don't have PF4 designs, and are also missing spacing:

image

The "remote viewer" details also leave something to be desired:

image

I'll re-do a build from a clean git to double-check.

**Comment by KKoukiou** [comment]
> I visually tested this now in a fedora-33 VM, and the buttons don't have PF4 designs, and are also missing spacing:
> 
> image
> 
> The "remote viewer" details also leave something to be desired:
> 
> image
> 
> I'll re-do a build from a clean git to double-check.

Hmm this looks like the CSS from the new console components in not applied at all. Let me know if it persists on a clean branch.

**Comment by martinpitt** [comment]
> I'll re-do a build from a clean git to double-check.

Done now, and it looks much better:

image

image

Sorry for the noise then! LGTM.

File: package.json (hunk 1 above)
Comment (KKoukiou) [reply package.json:11]: I had some problems with removing it, with unatisfied deps, I can try once again. Maybe after mergin this?

File: pkg/machines/components/vm/consoles/consoles.jsx [hunk 2]
```diff
@@ -41,67 +40,14 @@ const VmNotRunning = () => {
...
-};
-
 class Consoles extends React.Component {
     constructor (props) {
         super(props);
 
         this.state = {
-            consoleType: undefined,
             consoleDetail: undefined,
         };
```
Comment (skobyda) [inline pkg/machines/components/vm/consoles/consoles.jsx:49]: Sorry to jump in on review so late, but I see that this state property "consoleDetail" is not used anywehere, and this component Consoles could be a stateless component

**Review by skobyda — CHANGES_REQUESTED** [review]

File: pkg/machines/components/vm/consoles/consoles.jsx [hunk 3]
```diff
@@ -165,34 +92,34 @@ class Consoles extends React.Component {
...
-        default:
-            break;
-        }
-
-        return (<NoConsoleDefined />);
     }
 }
 Consoles.propTypes = {
     vm: PropTypes.object.isRequired,
-    config: PropTypes.object.isRequired,
```
Comment (skobyda) [inline pkg/machines/components/vm/consoles/consoles.jsx:195]: also config is still passed to Consoles component, but not used anywhere

**Review by skobyda — APPROVED** [review]
Since there will be a followup PR, and my comments are mostly only related to redundant state properties and component parameters (so nothing really serious, just a cleanup issues), I'm okay approving it now and doing cleanup in some other PR.

File: package.json (hunk 1 above)
Comment (martinpitt) [reply package.json:11]: Not important at all, fine for me to merge as-is (and I approved). Not sure about the issue @skobyda just found, if it's just harmless cleanup or a blocker.

**Comment by KKoukiou** [comment]
> Since there will be a followup PR, and my comments are mostly only related to redundant state properties and component parameters (so nothing really serious, just a cleanup issues), I'm okay approving it now and doing cleanup in some other PR.
> Also tried it out locally and works nice! Thanks for nice PR Katerina.

I will definitely send a followup because there is also the SerialConsole to port. I will address the two small issues in the followup! Thanks :)

**Pull request merged**